### DISCUSSION TITLE
Azure Push Notification on Android always adds silent notification to system tray

### DISCUSSION THREAD

**Discussion by jmichas** [shinyorg/shiny#1170] [Q&A] [answered]
I am not sure if this is a bug or not or if I am missing something.
If I have an empty `IPushDelegate` handler with no code in `OnReceived`, when the app on android is in the foreground, messages are entered into the silent notifications system tray. My message has both a `data` and `notification` object in the JSON such as:
````
{
    "data": {
        "correlationId": "81839",
        "silent": false
    },
    "notification": {
        "title": "Some title",
        "body": "Some body text!",
        "icon": "notification_icon",
        "image": ""
    },
    "android": {
        "priority": "high",
        "notification": {
            "channel_id": "MyChannel"
        }
    }
}
````
and I am sending it through the azure test console.

I am not sure how to prevent the notification from being put in the system tray's silent notifications.

The problem with this is that if I catch the notification and send it via the `NotificationManager` so it will display to the user, I now end up with two identical system tray notifications, one silent and one normal.

I am using Shiny 2.7 upgrading from 2.4.3. I feel like this behavior did not exist in the older version.

**Comment by jmichas** [comment, marked as answer]
Ok, so this might and probably is not an issue. In looking at my app settings on the device it seems there is a default "Notification" channel listed that is not present in my QA or Production versions. It was likely in there from a long ago version when I was first working on push notifications. But, I had no idea that the channels persisted across installs. I completely uninstalled my dev version before beginning this work and assumed that the settings like that would be erased. Is there an easy way for me to delete the channel? I want my dev to mimic my other version's settings so it equates to what real users have when installing the app.

**Reply by aritchie** [reply]
Channels definitely persist (especially on Android) across installs.  There is an IChannelManager that you can work with as needed.  It is automatically registered with push.
